### PR TITLE
Introduce extension: changes

### PR DESCRIPTION
**Pull request by TSMMark**
- [x] document in readme
- [x] test in production app
- [ ] release new minor version

### CODE REVIEW

**Comment by TSMMark** [comment]
[refactor ROLES/PERMISSIONS to use gourami `changes?`](https://trello.com/c/Fz4NyOmD/5160-refactor-roles-permissions-to-use-gourami-changes)

File: README.md [hunk 1]
```diff
@@ -175,6 +173,81 @@ class UpdateFishBowl < CreateFishBowl
...
+  include Gourami::Extensions::Changes
+
+  record(:page)
+  attribute(:widgets,
+    :type => :array,
+    :element_type => :string
+    :watch_changes => ->(new_value) {
+      did_change(:pro_widget, new_value.include?("pro"))
+      new_value.sort.uniq != page.widgets.sort.uniq
+    })
```
Comment (paulducsantos) [inline README.md:242]: indendation?

File: lib/gourami/attributes.rb [hunk 2]
```diff
@@ -23,39 +23,42 @@ def attribute(name, options = {}, &default_block)
...
 
-        case options[:type]
-        when :boolean
-          define_method(:"#{name}?") do
-            !!send(name)
+          case options[:type]
+          when :boolean
+            mixin.send(:define_method, :"#{name}?") do
+              !!send(name)
+            end
```
Comment (paulducsantos) [inline lib/gourami/attributes.rb:56]: do you need a case statement to just check if boolean?

**Review by paulducsantos — APPROVED** [review]

File: lib/gourami/attributes.rb (hunk 2 above)
Comment (TSMMark) [reply lib/gourami/attributes.rb:56]: I mean not really but leaves open for more

File: README.md (hunk 1 above)
Comment (TSMMark) [reply README.md:242]: what's wrong with it?

**Pull request merged**